### DISCUSSION TITLE
how do i search for an empty string in PowerShell?

### DISCUSSION THREAD

**Discussion by eruizc-dev** [BurntSushi/ripgrep#1713] [Q&A] [answered]
#### What version of ripgrep are you using?

```
ripgrep 12.1.1
-SIMD -AVC (compiled)
+SIMD +AVX (runtime)
```

#### How did you install ripgrep?

`sudo pacman -S ripgrep`

#### What operating system are you using ripgrep on?

 - OS: Manjaro Linux x86_64
 - Desktop: Plasma 5.19.5
 - Shell: pwsh PowerShell 7.0.3

#### Describe your bug.

I cannot perform an empty search using Powershel Core 7. This causes programs like `fzf` to fail when ran on powershell.

#### What are the steps to reproduce the behavior?

The following code produces `error: The following required arguments were not provided: <PATTERN>`

```ps1
rg ''
```

The use of `""` and `[string]::Empty` produces the same output

#### What is the expected behavior?

I expect RipGrep to match all lines in all files like it does with bash. In fact, you can try these two commands and you will see pwsh fail and bash succeed.

```bash
pwsh -c "rg ''"
```

```bash
bash -c "rg ''"
```

**Comment by BurntSushi** [comment]
Try `rg ""`.

**Comment by eruizc-dev** [comment]
> Try `rg ""`.

I forgot to specified I tried that and `[string]::Empty` and none of those worked

**Comment by BurntSushi** [comment]
I'm not sure then. My guess is that this is a PowerShell issue and not a ripgrep issue, but it will be a while before I'm able to spin up a Windows machine and test this for myself. Others are encouraged to weigh in here!

**Comment by mklement0** [comment, marked as answer]
Indeed, this is purely a PowerShell issue: empty-string arguments are simply _discarded_, regrettably; argument-passing to external programs has always been fundamentally broken in PowerShell, notably also with respect to handling embedded `"` chars.; backward-compatibility concerns have prevented a fix so far - see https://github.com/PowerShell/PowerShell/issues/1995

Workarounds:

```
PS> rg --% ""
```

This workaround is future-proof, but the use of `--%` has important side effects: see `--%`, the [stop-parsing symbol](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_Parsing).

_For now_, the following works too, but it _shouldn't_, and if https://github.com/PowerShell/PowerShell/issues/1995 ever gets fixed - there's talk of providing a fix via an _experimental feature_ now -  it will stop working:

```powershell
PS> rg '""'
```

**Reply by BurntSushi** [reply]
Amazing answer!!! Thank you so much. :-)
